.github/workflows/ci.yaml

```diff
@@ -4,16 +4,113 @@ on:
   workflow_call:
 
 jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
   unit-test:
     name: Unit tests
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
       - name: Install dependencies
         run: python3 -m pip install tox
       - name: Run tests
         run: tox -e unit
+  build-jenkins-agent-image:
+    name: Build jenkins agent image
+    runs-on: ubuntu-22.04
+    outputs:
+      image-id: ${{ steps.build.outputs.imageid }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build_oci_image
+        id: build
+        with:
+          dockerfile: jenkins-agent.Dockerfile
+          resource-name: jenkins-agent-image
+  integration-test-lxd-microk8s:
+    name: Integration tests (lxd and microk8s)
+    # [20220824] charmed-kubernetes/actions-operator@main does not work on
+    # ubuntu-22.04, hence using 20.04
+    runs-on: ubuntu-20.04
+    needs: [build-jenkins-agent-image, lint, unit-test]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: lxd
+      - name: Install microk8s
+        run: |
+          sudo snap install microk8s --classic
+          sudo microk8s status --wait-ready
+          sudo microk8s.enable dns hostpath-storage
```

Comment:
Could we enable the registry as well, do we need the additional step bellow?

Reply:
Thank you, fixed